src/golibmc.go

```diff
@@ -623,8 +965,13 @@ func (client *Client) DeleteMulti(keys []string) (failedKeys []string, err error
 }
 
 func (client *Client) getOrGets(cmd string, key string) (item *Item, err error) {
-	client.lock()
-	defer client.unlock()
+	cn, err := client.conn(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		client.putConn(cn)
+	}()
```

Comment:
why not "defer client.putConn(cn)" as above?

Reply:
Thanks, fixed.
https://github.com/douban/libmc/pull/57/commits/9e06a614a58d35db97f46e68336dc18439a17881